From 44e4a5a86d0a3506f9b5e459384952f0df3124f3 Mon Sep 17 00:00:00 2001
From: Seth Forshee <seth.forshee@canonical.com>
Date: Fri, 6 Jul 2018 09:08:42 -0500
Subject: [PATCH] FROMGIT: UBUNTU: SAUCE: (no-up): ASoC: Intel: bytcr-rt5660: Remove
 snd_soc_codec use for 4.18

This struct has been removed from 4.18. It was only used for
getting the struct device, so get that from the card instead.

Signed-off-by: Seth Forshee <seth.forshee@canonical.com>
---
 sound/soc/intel/boards/bytcr_rt5660.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/sound/soc/intel/boards/bytcr_rt5660.c b/sound/soc/intel/boards/bytcr_rt5660.c
index f0e3e4125f07..3003ea2c6319 100644
--- a/sound/soc/intel/boards/bytcr_rt5660.c
+++ b/sound/soc/intel/boards/bytcr_rt5660.c
@@ -197,11 +197,10 @@ static int byt_rt5660_init(struct snd_soc_pcm_runtime *runtime)
 {
 	struct snd_soc_card *card = runtime->card;
 	struct byt_rt5660_private *priv = snd_soc_card_get_drvdata(card);
-	struct snd_soc_codec *codec = runtime->codec;
 	int ret;
 
 	/* Request rt5660 GPIO for lineout mute control */
-	priv->gpio_lo_mute = devm_gpiod_get_index(codec->dev,
+	priv->gpio_lo_mute = devm_gpiod_get_index(card->dev,
 			"lineout-mute", 0, 0);
 	if (IS_ERR(priv->gpio_lo_mute)) {
 		dev_err(card->dev, "Can't find GPIO_MUTE# gpio\n");
-- 
2.17.1

